tests/test_unittest/test_gtc/test_npir_gen.py

```diff
@@ -0,0 +1,466 @@
+
+
+def test_full_computation_valid(tmp_path) -> None:
+    result = npir_gen.NpirGen.apply(
+        npir.Computation(
+            params=["f1", "f2", "f3", "s1"],
+            field_params=["f1", "f2", "f3"],
+            field_decls=[
+                FieldDeclFactory(name="f1"),
+                FieldDeclFactory(name="f2"),
+                FieldDeclFactory(name="f3"),
+            ],
+            vertical_passes=[
+                VerticalPassFactory(
+                    temp_defs=[],
+                    body=[
+                        npir.HorizontalBlock(
+                            body=[
+                                VectorAssignFactory(
+                                    left=FieldSliceFactory(name="f1", parallel_k=True),
+                                    right=npir.VectorArithmetic(
+                                        op=common.ArithmeticOperator.MUL,
+                                        left=FieldSliceFactory(
+                                            name="f2", parallel_k=True, offsets=(-2, -2, 0)
+                                        ),
+                                        right=FieldSliceFactory(
+                                            name="f3", parallel_k=True, offsets=(0, 3, 1)
+                                        ),
+                                    ),
+                                ),
+                            ],
+                        ),
+                    ],
+                ),
+                VerticalPassFactory(
+                    lower=common.AxisBound.from_start(offset=1),
+                    upper=common.AxisBound.from_end(offset=-3),
+                    direction=common.LoopOrder.BACKWARD,
+                    temp_defs=[],
+                    body=[
+                        npir.HorizontalBlock(
+                            body=[
+                                VectorAssignFactory(
+                                    left__name="f2",
+                                    right=npir.VectorArithmetic(
+                                        op=common.ArithmeticOperator.ADD,
+                                        left=FieldSliceFactory(name="f2", parallel_k=False),
+                                        right=FieldSliceFactory(
+                                            name="f2", parallel_k=False, offsets=(0, 0, 1)
+                                        ),
+                                    ),
+                                ),
+                            ],
+                        )
+                    ],
+                ),
+            ],
+        ),
+        field_extents={
+            "f1": Extent([(0, 0), (0, 0)]),
+            "f2": Extent([(-2, 0), (-2, 0)]),
+            "f3": Extent([(0, 0), (0, 3)]),
+        },
+    )
+    print(result)
+    mod_path = tmp_path / "npir_gen_1.py"
+    mod_path.write_text(result)
+    import sys
+
+    sys.path.append(str(tmp_path))
+    import npir_gen_1 as mod
+    import numpy as np
```

Comment:
Should the sys and numpy imports go at the top of the file?

Reply:
I think I'd prefer that, yes